lib/types.nix

```diff
@@ -352,6 +352,25 @@ rec {
       functor = (defaultFunctor name) // { wrapped = [ t1 t2 ]; };
     };
 
+    coercedTo = coercedType: coerceFunc: finalType: mkOptionType rec {
```

Comment:
Assert that `coercedType` does not contain any submodule by checking that `getSubmodule` is `null`.